google-cloud-bigquery/lib/google/cloud/bigquery/service.rb

```diff
@@ -564,9 +589,10 @@ def execute
 
           protected
 
-          def retry? result, current_retries #:nodoc:
+          def retry? err, current_retries #:nodoc:
+            return true if err.is_a? JSON::ParserError # See #5180
```

Comment:
Shouldn't we check `current_retries` first? Otherwise it could retry infinitely if it's always a ParserError

Reply:
gulp, thanks for catching this!

Reply:
Fixed, with unit test coverage.